server/factory_event_server.py

```diff
@@ -0,0 +1,8 @@
+from server.server_event import AcceptChallenge
+
+
+class FactoryServerEvent(object):
+    @staticmethod
+    def get_event(data, client):
+        if data.get('action') == 'accept_challenge':
+            return AcceptChallenge(data, client)
```

Comment:
You could do something more like. Keep in mind that these constants shouldn't be placed here in real code
```
ACCEPT_CHALLENGE = 'accept_challenge'

CHALLENGES = {
    ACCEPT_CHALLENGE: AcceptChallenge,
}

class FactoryServerEvent(object):
    @staticmethod
    def get_event(data, client):
        challenge = CHALLENGE.get(data.get('action'), some_default_value)
        return challenge(data, client)
```